### PR TITLE
Add json export

### PR DESCRIPTION
**Pull request by vjanelle**

### CODE REVIEW

**Comment by vjanelle** [comment]
@ripienaar works (I think) on RPM.  Tests need work though.

File: .rubocop.yml [hunk 1]
```diff
@@ -158,7 +152,7 @@ Performance/RegexpMatch:
 Style/SafeNavigation:
   Enabled: false
 
-Lint/RescueWithoutErrorClass:
```
Comment (ripienaar) [inline .rubocop.yml]: this all should only be needed if you somehow updated rubocop - then you'll have a bad time anyway :)

File: lib/mcollective/pluginpackager.rb [hunk 2]
```diff
@@ -1,5 +1,8 @@
 module MCollective
   module PluginPackager
+
+    class GenerateJsonDirNotFoundError < EncodingError
+    end
```
Comment (ripienaar) [inline lib/mcollective/pluginpackager.rb]: surprised rubocop liked this, there should be a blank line below and none above I thought

File: lib/mcollective/pluginpackager.rb [hunk 3]
```diff
@@ -13,6 +16,39 @@ def self.[](klass)
...
+          end
+
+          File.open(json_file, "w") do |jddl|
+            jddl.print(JSON.pretty_generate(data))
+          end
+
+          plugin.packagedata[:common][:files] << "agent/%s.json" % agent_name
+        end
+      # else
+      #  raise(GenerateJsonDirNotFoundError) # Need to Mock this out
```
Comment (ripienaar) [inline lib/mcollective/pluginpackager.rb]: please remove if not used (and then the declaration of the exception too)

File: lib/mcollective/pluginpackager.rb [hunk 4]
```diff
@@ -13,6 +16,39 @@ def self.[](klass)
...
+      end
+
```
Comment (ripienaar) [inline lib/mcollective/pluginpackager.rb]: no blank line

File: lib/mcollective/pluginpackager/debpackage_packager.rb [hunk 5]
```diff
@@ -35,12 +35,14 @@ def create_packages
           @build_dir = File.join(@tmpdir, "#{@package_name}_#{@plugin.metadata[:version]}")
           Dir.mkdir(@build_dir)
 
+
```
Comment (ripienaar) [inline lib/mcollective/pluginpackager/debpackage_packager.rb]: no blank line

File: spec/unit/mcollective/pluginpackager_spec.rb [hunk 6]
```diff
@@ -151,5 +151,6 @@ module MCollective
                           {:name => 'rspec_debian', :version => '2.0'}]
       end
     end
+
```
Comment (ripienaar) [inline spec/unit/mcollective/pluginpackager_spec.rb]: no blank dir

**Comment by vjanelle** [comment]
@ripienaar I think I've switched to another process vs making RPMs, it was getting kinda unwieldy.

**Comment by vjanelle** [comment]
(Happy to finish this, but my use case is kinda dwindling)

**Comment by ripienaar** [comment]
yeah i am also looking to retire these ways of building modules so I'll close this

**Pull request closed without merging**